### PR TITLE
Install necessary packages during role execution

### PR DESCRIPTION
**Pull request by timflannagan**

### CODE REVIEW

File: tasks/fs-default.yml [hunk 1]
```diff
@@ -2,6 +2,24 @@
 - name: Stat the final device file
   include_tasks: stat_device.yml
 
+- name: Install xfsprogs for xfs file system type
+  package:
+    name: xfsprogs
+    state: latest
+  when: fs_type == "xfs" or fs_type == ""
```
Comment (dwlehman) [inline tasks/fs-default.yml]: This one should be only if fs_type is "xfs".

File: tasks/lv-default.yml [hunk 2]
```diff
@@ -1,4 +1,10 @@
 ---
+- name: Install LVM2 commands as needed
+  package:
+    name: lvm2
+    state: latest
+  when: device_type != "disk"
```
Comment (dwlehman) [inline tasks/lv-default.yml]: Only when device_type is "lvm".

File: tasks/vg-default.yml [hunk 3]
```diff
@@ -52,6 +52,12 @@
 #
 # Configure the VG
 #
+- name: Install LVM2 commmands as needed
+  package:
+    name: lvm2
+    state: latest
+  when: device_type != "disk"
```
Comment (dwlehman) [inline tasks/vg-default.yml]: Same here: only when device_type is "lvm".

**Review by dwlehman — COMMENTED** [review]
The only other thing I would suggest is to use 'present' everywhere you used 'latest'. We don't really want to trigger random/unexpected package upgrades if we already have what we need.

**Comment by timflannagan** [comment]
Updated with the suggested changes

File: tasks/vg-default.yml [hunk 4]
```diff
@@ -55,8 +55,8 @@
 - name: Install LVM2 commmands as needed
```
Comment (dwlehman) [inline tasks/vg-default.yml]: This has to be installed before we can run the 'lvs' command at the top.

**Review by dwlehman — COMMENTED** [review]
Looks good otherwise.

File: tasks/vg-default.yml (hunk 4 above)
Comment (timflannagan) [reply tasks/vg-default.yml]: Fixed

Comment (timflannagan) [reply tasks/vg-default.yml]: Is the command 'pvs' in the lvm2 package?

Comment (dwlehman) [reply tasks/vg-default.yml]: Yes, so package installation will need to go at the top of that file.

Comment (timflannagan) [reply tasks/vg-default.yml]: Updated.

**Review by dwlehman — APPROVED** [review]
Looks good!

**Pull request merged**